Migrate `SitePluginModel` to `Room`

wzieba opened the pull request:
<!-- Remember about a good descriptive title. -->

<!-- Id number of the Linear issue this PR addresses, e.g., WOOMOB-373. -->

Closes: [AINFRA-547: Migrate `SitePluginModel`](https://linear.app/a8c/issue/AINFRA-547/migrate-sitepluginmodel)

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

This PR migrates `SitePluginModel` from `WellSql` to `Room`.

### Test Steps
<!-- Describe how to test your changes. Include only what’s needed for the reviewer to validate the behavior.
For new features, outline the main user flow or goal rather than every tap or click — the reviewer should be able to complete the flow naturally.
If applicable, mention key devices, scenarios, or edge cases to verify. -->

#### Plugins list
1. Go to settings > Plugins > Installed Plugins - verify all plugins are listed like before

#### Plugin installation
1. You'll likely need a new site: I've used Jurrasic
2. Go to settings
3. Tap on Install Jetpack
4. Log in with a WordPress account **⚠️ BUT** do not use A8c account - it won't work with it
5. Make sure that Jetpack installation is smooth and Jetpack was installed

### Images/gif
<!-- Include before and after images or gifs when appropriate. -->
<img width="1589" height="234" alt="Screenshot 2025-12-19 at 15 51 22" src="https://github.com/user-attachments/assets/a0040b06-7014-4324-87f7-5489a243a267" />



- [ ] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

wzieba committed:
Remove unused code related to SitePlugin model

wzieba committed:
Add Room infrastructure for SitePlugin migration

wzieba committed:
Update WooCommerceStore to use SitePluginDao

wzieba committed:
Add and update tests for Room-based SitePlugin DAO

All tests use the new DAO methods (upsert, replaceAllSitePlugins) and
proper coroutine testing with runTest instead of runBlocking.

wzieba committed:
Remove WellSql-based PluginSqlUtils infrastructure

wzieba committed:
Introduce a `createTestSitePlugin` test utility

wzieba committed:
Add `WellSql` migration to drop `SitePluginModel` table

wzieba commented on libs/fluxc-plugin/src/main/kotlin/org/wordpress/android/fluxc/store/WooCommerceStore.kt:
```diff
@@ -173,23 +173,32 @@ open class WooCommerceStore @Inject internal constructor(
 
     fun getActiveSitePlugin(site: SiteModel, plugin: WooPlugin): SitePluginModel? {
-        return pluginSqlUtils.getActiveSitePluginByName(site, plugin.pluginName)
+        return runBlocking {
+            val allPlugins = sitePluginDao.getActiveSitePlugins(site.localId())
+            val extractedRequestedPluginName = plugin.pluginName.substringAfterLast('/')
+            allPlugins.firstOrNull { sitePlugin ->
+                val extractedPluginName = sitePlugin.name.substringAfterLast('/')
+                extractedPluginName == extractedRequestedPluginName
+            }
+        }
```
Logic from: https://github.com/woocommerce/woocommerce-android/pull/15098/changes#diff-d00aa3e672c503d65df49e81ce1f69c1e50c69a2b709b4d2a22247612e2b3ec7L84-L98

samiuelson committed:
Merge branch 'trunk' into migrate_site_plugin_model

samiuelson reviewed (approved):
The PR tests good 👍 I left a couple of unresolved comments from Copilot to verify.

wzieba commented on libs/fluxc/src/main/java/org/wordpress/android/fluxc/persistence/SitePluginDao.kt:
```diff
@@ -0,0 +1,46 @@
+import org.wordpress.android.fluxc.model.plugin.SitePluginModel
+
+@Dao
+abstract class SitePluginDao {
```
It's by design and consistent with already more than 30 migrations 👍. I don't see that this comment will help to navigate.

wzieba commented on libs/fluxc/src/testFixtures/kotlin/org/wordpress/android/fluxc/wp/site/SitePluginFixtures.kt:
```diff
@@ -0,0 +1,22 @@
+        siteId: Int = 1,
+        name: String = "",
+        version: String = "1.0.0",
+        slug: String = name.lowercase().replace(" ", "-"),
```
This is `testFixtures` directory, not a production code. It's okay to make fake assumptions here.

wzieba commented on libs/fluxc/src/main/java/org/wordpress/android/fluxc/model/plugin/SitePluginModel.kt:
```diff
@@ -0,0 +1,19 @@
+    tableName = "SitePluginEntity",
+    primaryKeys = ["siteId", "slug"]
+)
+data class SitePluginModel(
+    val siteId: LocalId,
+    val name: String,
+    val version: String,
+    val slug: String,
+    val authorName: String,
+    @ColumnInfo(name = "isActive")
+    val isActive: Boolean
```
These fields were removed intentionally as they were unused. 5bd8f76a7548562437316e9c900d0d795c882ac3

wzieba commented on libs/fluxc-plugin/src/main/kotlin/org/wordpress/android/fluxc/network/rest/wpcom/wc/system/WCSystemPluginResponse.kt:
```diff
@@ -30,14 +30,12 @@ data class WCSystemPluginResponse(
+        siteId = LocalId(siteId),
+        name = this.plugin?.substringBeforeLast(".") ?: "",
+        version = this.version ?: "",
+        slug = this.plugin?.substringBeforeLast("/") ?: "",
```
We don't observe problems with this logic in production and I simply migrated exsisting code:

https://github.com/woocommerce/woocommerce-android/blob/98fa016beeacb47d07b99e5658d1cc0c1d7eda36/libs/fluxc-plugin/src/main/kotlin/org/wordpress/android/fluxc/network/rest/wpcom/wc/system/WCSystemPluginResponse.kt#L39

wzieba committed:
Include `name` in primary key of `SitePluginEntity`

It's probably possible that `slug` could be empty. For such cases, `name` should be also taken into account for primary key. If both `name` and `slug` would be empty - then we can assume something is really wrong and not a valid state

wzieba committed:
Replace mocks of `SitePluginModel` with fake

`SitePluginModel` is now a data class, so mocking it causes lint error

wzieba commented on libs/fluxc/src/main/java/org/wordpress/android/fluxc/model/plugin/SitePluginModel.kt:
```diff
@@ -0,0 +1,19 @@
+
+@Entity(
+    tableName = "SitePluginEntity",
+    primaryKeys = ["siteId", "slug"]
```
I've added `name` to the key in 654bef2 - just in case, as it's true that we should be very cautious with API responses 👍

wzieba commented:
Thank you @samiuelson for review! I checked the opened comments, replied to them and added one fix in 654bef2 with explanation.

wzieba committed:
Update unit tests to reflect `name` as part of primary key

The primary key change was introduced in
654bef2. I assume here that both name and slug of plugin don't change
